src/controllers/Order.controller.ts

```diff
@@ -13,24 +13,20 @@ import { Orderinterface } from "../interfaces";
 export const createOrder = async (req: Request, res: Response) => {
   try {
     const userId = req.user.id;
-    const items = await getCartItemsService(userId);
-    if (!items) {
+    const items: Orderinterface[] = await getCartItemsService(userId);
+    if (!items[0]) {
       return res.status(400).json({
         message: "No product available To make order",
         status: 400,
         success: false,
       });
     }
-    const data: Orderinterface = {
-      QUANTITY: items[0].quantity,
-      PRODUCT: items[0].product,
-    };
-    const order = await createOrderService(userId, data);
+    const order = await createOrderService(userId, items);
     res.status(201).json({
-      data: order,
       message: "Order created",
       status: 201,
       success: true,
+      data: order,
     });
```

Comment:
I some few insights into this section.

**When creating an order**

1. I think when creating an order, one should first have access to the total amount that is the cart, and I am not seeing it anywhere here.
2. Speaking of the cart, there is already a logic to grab the cart items of the logged-in, and a way to calculate the total right way.

My suggestion is this.

- I think you can find a way to make that logic of making those two functions re-usable inside there own function. i.e - retrieving the cart using the `viewCartService` and `calculateCartTotal`. If you get stuck here, please let me know and we try together.
- Two, I think you should be mindful of how placing an order will work. It will take the information of the product, it's price, it's quantity, and all that information will re-present the order, so that when a user reecieves the receit, they must see that info.
e.g 
```
Order Number. The order number will be the reference number for the order made. Should there be issues regarding the orders such as payments or if there is missing information, it will be easy for the customer to let you know what their issues are.

Order Date. This is basically where the date when the order was created will reflect.

Sold to. This is where the name and details of the customer will reflect.

Name of customer or company representative that will receive the orders (surname, and given_name).
Complete address (house number/building name, street name, city, state, and zip code).

Date of Shipment. This is where the date when the items were shipped is indicated. If there are items that come from different suppliers, there might also be a possibility that these items will be shipped on different dates.

Order details. This is the part of the order where the details about the orders will reflect. You can find the following in this part of the order receipt:

    Item code or item name
    Item description
    Quantity per item
    Price per unit item

Bill details. This pertains to the prices that are related to the order. It is usually broken down into the following items:

    Total amount to be paid
    Shipping and handling fee
    Amount paid per item, and how many items.

Payment details. This will consist of the following:

    Payment method/s accepted such as check, cash, COD (cash on delivery), deposit, debit/credit, etc.
    Amount billed

Signature. Your signature will serve as the [order confirmation](https://www.sampletemplates.com/sample-forms/order-confirmation-form.html) that the orders have been received.

```